Tests: USB: Fix Python 3 compatibility

fkjagodzinski opened the pull request:
### Description

<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->
This is a follow-up to #11520.

Use integer division explicitly in basic test.
Convert bytes to str for Python 3 in serial test.

**After the update, this patch also bumps the version of `six` module to 1.12.0.**

### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [x] Test update
    [ ] Breaking change

### Reviewers

<!--
    Optional
    Request additional reviewers with @username
-->
@OPpuolitaival @madchutney 

### Release Notes

<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

madchutney commented on TESTS/host_tests/usb_device_serial.py:
```diff
@@ -64,6 +64,16 @@ def usb_serial_name(serial_number):
     return None
 
 
+if sys.version_info < (3,):
```
Can this be done with https://pypi.org/project/six/

fkjagodzinski replied:
@madchutney thanks for pointing this out. 👍 I haven't used this module before.

[`six.ensure_str()`](https://six.readthedocs.io/#six.ensure_str) looks exactly like what I need here. Too bad we currently use an older version of this module. https://github.com/ARMmbed/mbed-os/blob/daafb950d96e2e72027f72f1dbdfdd6333cf7fbc/requirements.txt#L8

`six.ensure_str()` was added in v1.12.0 (https://github.com/benjaminp/six/commit/db3d0d678b422614d5de0f9ed76a2112a28c7b19). @madchutney, @0xc0170 can we ask someone to bump the `six` version?

OPpuolitaival replied:
Why this cannot be inside function?

fkjagodzinski replied:
@OPpuolitaival, either inside or outside of the function definition; doesn't make any difference for me. I think I left it this way after seeing http://python3porting.com/problems.html#nicer-solutions.

OPpuolitaival replied:
No big matter for me

0xc0170 replied:
> six.ensure_str() was added in v1.12.0 (benjaminp/six@db3d0d6). @madchutney, @0xc0170 can we ask someone to bump the six version?

I've just noticed this question here. We can do it if there is a need, with full testing of course.

fkjagodzinski committed:
Bump the six (Python module) version to 1.12.0

This version provides a few handy functions: six.ensure_binary,
six.ensure_text, and six.ensure_str.

fkjagodzinski committed:
Tests: USB: Fix Python 3 compatibility

Use integer division explicitly in basic test.
Convert bytes to str for Python 3 in serial test.
